pkg/pipeline/input/sdk/synchronizer.go

```diff
@@ -116,37 +131,47 @@ func (c *synchronizer) onRTCP(packet rtcp.Packet) {
 				t := c.syncInfo[ssrc]
 				if diff := ntpStart.Sub(minNTPStart); diff != 0 {
 					t.Lock()
-					t.ptsDelay += int64(diff)
+					t.ptsOffset += int64(diff)
 					t.Unlock()
 				}
 			}
 		} else {
-			t := c.syncInfo[pkt.SSRC]
-			pts, _ := t.getPTS(int64(pkt.RTPTime))
-			expected := mediatransportutil.NtpTime(pkt.NTPTime).Time().Sub(c.ntpStart)
-			if pts != expected {
-				diff := expected - pts
-				if diff > largePTSDiff || diff < -largePTSDiff {
-					logger.Warnw("high pts drift", nil, "trackID", t.trackID, "pts", pts, "diff", diff)
+			c.syncInfo[pkt.SSRC].onSenderReport(pkt, c.ntpStart)
+		}
+	}
+}
+
+// onSenderReport handles pts adjustments for a track
+func (t *syncInfo) onSenderReport(pkt *rtcp.SenderReport, ntpStart time.Time) {
+	pts, _ := t.getPTS(int64(pkt.RTPTime))
+	expected := mediatransportutil.NtpTime(pkt.NTPTime).Time().Sub(ntpStart)
+	if pts != expected {
+		diff := expected - pts
+		apply := true
+		t.Lock()
+		if absGreater(diff, largePTSDrift) {
+			logger.Warnw("high pts drift", nil, "trackID", t.trackID, "pts", pts, "diff", diff)
+			if absGreater(diff, massivePTSDrift) {
```

Comment:
With this new algorithm, it looks like we'll still fail the export if there is a jump in the timestamp somewhere that lasts more than 1 RTCP report?

Reply:
If they're consistent, then I think a large drift could be possible/valid in a situation where the video track begins muted and we don't know the framerate